src/metrics.cpp

```diff
@@ -103,6 +104,30 @@ double GetLocalSolPS()
     return miningTimer.rate(solutionTargetChecks);
 }
 
+int EstimateNetHeightInner(int height, int64_t tipmediantime,
+                           int heightLastCheckpoint, int64_t timeLastCheckpoint,
+                           int64_t targetSpacing)
+{
+    // We average the target spacing with the observed spacing to the last
+    // checkpoint, and use that to estimate the current network height.
+    int medianHeight = height - CBlockIndex::nMedianTimeSpan / 2;
+    double checkpointSpacing = (double (tipmediantime - timeLastCheckpoint)) / (medianHeight - heightLastCheckpoint);
```

Comment:
Provided we have floating point non-stop mode enabled (via [feholdexcept](http://www.cplusplus.com/reference/cfenv/feholdexcept/) or [fesetexceptflag](http://www.cplusplus.com/reference/cfenv/fesetexceptflag/)). I assumed we did because we use infinities in the fee handling code.